### DISCUSSION TITLE
Does Datafusion support versions for delta table

### DISCUSSION THREAD

**Discussion by krishna-prasad-s** [apache/datafusion#10041] [Q&A] [answered]
I'm working on a merge activity where I want to replicate the checkpoint behaviour of structured spark streaming.
I am working with delta tables and I have access to table versioning.
I am thinking to apply the version number as an alternative to structured spark streaming's checkpoint.

Does datafusion support creating an queriying between versions?
for e.g.

`    pub async fn get_df_from_version(&mut self, table_name: &str, version: i64) -> DataFrame {
        let sql = &format!("SELECT * FROM {} VERSION BETWEEN {} AND current", table_name, version);
        self.ctx.sql(sql).await.unwrap()
    }
`

**Comment by Jefffrey** [comment, marked as answer]
Some related discussion: https://github.com/apache/arrow-datafusion/issues/7292

Looks like some sqlparser changes were merged to allow time travel syntax, albeit for specific dialects.
